### ISSUE TITLE
.7 RC: no clickable link provided for draft connect on Linux

### ISSUE THREAD

**Issue by squillace** [Azure/draft#327]
Hi all, on Linux, with the rc 2 of .7, I get this. Because the "localhost:36165" doesn't have http:// in front of it, I can't click it to open the site. :-( I would prefer either we add that protocol prefix so that natural terminal clicks are offered, or else even just spawn the site. If I chose `Draft connect` it's fairly clear that the default browser should open it, right?

![image](https://user-images.githubusercontent.com/1422718/30408935-d2f18a14-98b6-11e7-8b5a-917ba3ea9c40.png)

**Comment by squillace** [comment]
Elsewise... what an amazing improvement. AMAZING!!!

**Comment by fibonacci1729** [comment]
1. I agree with the adding the "http(s)" prefix.
2. Would an optional `--open` flag (default to false) be sufficient?

**Comment by bacongobbler** [comment]
I completely agree this would be a fantastic feature for web apps. The problem is that Draft does not deploy strictly web-facing applications, as some apps (like draftd or gitlab) are not HTTP based. In fact they're gRPC or SSH endpoints.

In kubernetes the port type can be either "TCP" (the default, such as all the example apps) or "UDP". We could attach that to the local address displayed on `draft connect`, but that still wouldn't be very useful to the webapp developer. That would get us closer to the target requirement, though.

**Comment by radu-matei** [comment]
Reiterating through @squillace's comments on #422 (and posting in the correct issue...):

> 1. We could detect port 80 and prefix based on that.
> 2. We could spawn the location, but then add a "if it's a web app, click http://localhost:xxxx; otherwise, open the endpoint using the appropriate tool."
> 3. We could add some metadata in the draft.toml file for this, and prompt the user to switch it off/on?
> 4. ....uh....
I'm inclined to go with 2 or 3... thoughts?

**Comment by bacongobbler** [comment]
I wouldn't mind option 2!

**Comment by bacongobbler** [comment]
I'm leaning more and more towards just making the default protocol for `draft connect` set to `http://`. Even if the underlying application protocol isn't http, the user knows that and will handle the connection string accordingly. It's a nice enhancement for users who DO write web applications with Draft (which one could argue is probably the large majority of our userbase).
